### PR TITLE
Added condition that allow to use repository in console mode

### PR DESCRIPTION
**Pull request by k3rnel**
**BUGS:**

>Please describe the issue that you solved if its not filed.

Fixed error when trying to use update method in console mode.

### CODE REVIEW

**Comment by k3rnel** [comment]
Cmon guys, 10 days one condition

**Comment by devansh-webkul** [comment]
Hi @k3rnel,

Can you show me a screenshot of the error?

Because there are 4 more types also and one abstract class as well. So concerned is not related to the configurable product.

**Comment by k3rnel** [comment]
Okay I found a solution. Where can I define some function, that we can use in entire application?

**Comment by devansh-webkul** [comment]
Hi @k3rnel,

First of all, I want to know what error are you facing and the steps to reproduce that error.

And the second one is for the product changes you need to review all the classes in this path `packages/Webkul/Product/src/Type`.

The main class is `AbstractType` as this class is inherited to all types and some methods are overridden as well.

**Comment by k3rnel** [comment]
Create some artisan command and try to use ProductRepository in there. Try to update some configurable product. In my case its seems like this:

```php
/** @var \Webkul\Product\Models\Product $product */
  $product = $this->productRepository->create([
      'type'                => 'configurable',
      'attribute_family_id' => $attributeFamily->id,
      'sku'                 => $sku,
      'super_attributes'    => $superAttributes,
      'family'              => $attributeFamily->id,
  ]);
```
```php
$this->productRepository->update([
      'channel'           => 'Store',
      'name'              => $name,
      'url_key'           => Str::slug($name),
      'status'            => '1',
      'guest_checkout'    => '1',
      'short_description' => '<p>'.$this->faker->text(150).'</p>',
      'description'       => '<p>'.$this->faker->text(500).'</p>',
      'images'            => [
            'image_1' => ''
      ],
      'meta_title'        => '',
      'meta_keywords'     => '',
      'meta_description'  => '',
      'locale'            => 'en',
      'channels'          => $channels->pluck('id')->toArray(),
      'variants'          => $this->prepareVariantsData($product)
], $product->id);
```
using request in non-request env

File: packages/Webkul/Product/src/Type/Configurable.php [hunk 1]
```diff
@@ -96,7 +96,7 @@ public function update(array $data, $id, $attribute = "id")
     {
         $product = parent::update($data, $id, $attribute);
 
-        if (request()->route()->getName() != 'admin.catalog.products.massupdate') {
+        if (app()->runningInConsole() || request()->route()->getName() != 'admin.catalog.products.massupdate') {
```
Comment (devansh-webkul) [inline packages/Webkul/Product/src/Type/Configurable.php:99]: There is no need for this check, just try with this,

~~~php
$route = request()->route() ? request()->route()->getName() : '';
~~~

Take a reference from `AbstractType` class.

**Review by devansh-webkul — CHANGES_REQUESTED** [review]

File: packages/Webkul/Product/src/Type/Configurable.php [hunk 2]
```diff
@@ -96,7 +96,7 @@ public function update(array $data, $id, $attribute = "id")
     {
         $product = parent::update($data, $id, $attribute);
 
-        if (request()->route()->getName() != 'admin.catalog.products.massupdate') {
+        if (!request()->route() || request()->route()->getName() != 'admin.catalog.products.massupdate') {
```
Comment (devansh-webkul) [inline packages/Webkul/Product/src/Type/Configurable.php:99]: Not like that, check this,
https://github.com/bagisto/bagisto/blob/2b4ca622b43910536b8441fce6a5e5f66a221774/packages/Webkul/Product/src/Type/AbstractType.php#L212

~~~php
$route = request()->route() ? request()->route()->getName() : "";

if ($route !== 'admin.catalog.products.massupdate') {
~~~

**Review by devansh-webkul — CHANGES_REQUESTED** [review]
There are some changes needed and also update this class also,

- [x] packages/Webkul/Product/src/Type/Bundle.php's update method
- [x] packages/Webkul/Product/src/Type/Downloadable.php's update method
- [x] packages/Webkul/Product/src/Type/Grouped.php's update method

**Comment by k3rnel** [comment]
I did what you said. But for the future, we need to implement the most universal way to solve all the problems associated with checking the name of the route. Or I can create another pull request for this. Can you just tell me where I can deploy some function/method that I can use in all packages

**Comment by k3rnel** [comment]
> There are some changes needed and also update this class also,
> 
> * [ ]  packages/Webkul/Product/src/Type/Bundle.php's update method
> * [ ]  packages/Webkul/Product/src/Type/Downloadable.php's update method
> * [ ]  packages/Webkul/Product/src/Type/Grouped.php's update method

Okay

**Comment by devansh-webkul** [comment]
> I did what you said. But for the future, we need to implement the most universal way to solve all the problems associated with checking the name of the route. Or I can create another pull request for this. Can you just tell me where I can deploy some function/method that I can use in all packages?

Well, this not necessarily needed because everywhere it's already handled. Because it obvious in the terminal you always get null.

**Comment by devansh-webkul** [comment]
Hi @k3rnel,

Just one last thing, update your branch with the master as it is 46 commits behind.

**Comment by k3rnel** [comment]
Hi @devansh-webkul ,

Merged :)

**Review by devansh-webkul — APPROVED** [review]

**Pull request merged**